rust-runtime/smithy-http/src/byte_stream.rs

```diff
@@ -154,6 +156,38 @@ mod bytestream_util;
 ///     }
 ///     ```
 ///
+/// ## Getting data into a ByteStream
+/// ByteStreams can be created in one of three ways:
+/// 1. **From in-memory binary data**: ByteStreams created from in-memory data are always retryable. Data
+/// will be converted into `Bytes` enabling a cheap clone during retries.
+///     ```rust
+///     use bytes::Bytes;
+///     use smithy_http::byte_stream::ByteStream;
+///     let stream = ByteStream::from(vec![1,2,3]);
+///     let stream = ByteStream::from(Bytes::from_static(b"hello!"));
+///     ```
+///
+/// 2. **From a file**: ByteStreams created from a path can be retried. A new file descriptor will be opened if a retry occurs.
+///     ```rust
+///     use smithy_http::byte_stream::ByteStream;
+///     let stream = ByteStream::from_path("big_file.csv");
+///     ```
+///
+/// 3. **From an `SdkBody` directly**: For more advanced / custom use cases, a ByteStream can be created directly
+/// from an SdkBody. **When created from an SdkBody, care must be taken to ensure retriability.** An SdkBody is retriable
```

Comment:
Nit: retriable -> retryable to be consistent with the function name

Reply:
```suggestion
/// from an SdkBody. **When created from an SdkBody, care must be taken to ensure retriability.** An SdkBody is retryable
```